File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/GettingStarted.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.*;
import dev.typr.foundations.connect.*;

@SuppressWarnings("unused")
public class GettingStarted {
  // start
  Transactor tx = SingleConnectionDataSource.create(DuckDbConfig.inMemory().build()).transactor();

  int result = Fragment.of("SELECT 42").queryExactlyOne(DuckDbTypes.integer).transact(tx);
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/GettingStarted.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.Fragment.sql
import dev.typr.foundationssc.connect.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object GettingStarted:
  // start
  val tx: Transactor =
    SimpleDataSource
      .create(DuckDbConfig.inMemory().build())
      .transactor()

  val result: Int = sql"SELECT 42"
    .queryExactlyOne(DuckDbTypes.integer)
    .transact(tx)
  // stop
